### DISCUSSION TITLE
Vitest not calculating __css well

### DISCUSSION THREAD

**Discussion by AmirTugi** [chakra-ui/chakra-ui#8883] [General]
### Describe the bug

I am using `@chakra-ui/react` with `@emotion/react` and running `vitest` with `@testing-library`.
I am able to run my application smoothly, and when I try to run my unit tests, the calculated styles seem to go awry.

The issue is surfacing when an element that should be visible (using `expect().toBeVisible()`), and it is not.

<img width="422" alt="image" src="https://github.com/vitest-dev/vitest/assets/8065975/2cbd855b-ebc5-41ad-a119-9fa15116488a">

I have this styling definitions for Chakra:
```tsx
const theme = {
  components: {
    Cell: {
      baseStyle: {
        backgroundColor: "teal",
        opacity: 1,
        _hover: {
          opacity: 0, // <-- This seems to be applied all the time in vitest.
        },
      },
    },
  },
};
```
When I print the computed style of the opacity while running the test - I see that the "hover" style is taking precedence over the default one.

Something does not add up with the fact that the application works with `Vite` but the tests using `vitest` are not applying the styling well.

I have tried changing `jsdom` to `happy-dom` and run in the [browser mode](https://vitest.dev/guide/browser.html) to no avail.
I have also tried adding the emotion babel plugin and adding `jsxImportSource: "@emotion/react",` but that still doesn't fix the tests.

After a couple of days of research, I came to a conclusion that is must be a something that fails in vitest.

Versions:
`@chakra-ui/react`: `1.8.9`,
`@emotion/react`: `11.11.1`,
`vitest`: `0.34.6`,
`vite`: `4.5.0`,
`react`: `17.0.2`,
`@testing-library/jest-dom`: `6.1.4`,
`@testing-library/react`: `12.1.2`,

### Reproduction

I have a [codesandbox](https://codesandbox.io/p/sandbox/confident-fast-67vlvw?file=%2Ftsconfig.json%3A9%2C37&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clozs6ekd000a3b6huws40ft8%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clozs6ekd00033b6hzhkhigth%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clozs6ekd00073b6h2nn3j5ox%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clozs6ekd00093b6h042la2ar%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B57.833957888181416%252C42.166042111818584%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clozs6ekd00033b6hzhkhigth%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00033b6hzhkhigth%2522%252C%2522activeTabId%2522%253A%2522clozzhs1c0c2u3b6hredvibof%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clozs6ekd00023b6hl95p6b1y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clozu1jqi00e73b6h288ih2nl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clozu5xvm02233b6henjbx4ur%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FCell.tsx%2522%252C%2522id%2522%253A%2522clozu86fm038f3b6h92zw1jo7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.test.tsx%2522%252C%2522id%2522%253A%2522clozz4max09tt3b6hwb9ytrvw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fvite.config.ts%2522%252C%2522id%2522%253A%2522clozzhs1c0c2u3b6hredvibof%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FProvider.tsx%2522%252C%2522id%2522%253A%2522clp14u11g0iz63b6h999g5a8n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clp15336000033b6i7vnhz7oj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clozs6ekd00093b6h042la2ar%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00093b6h042la2ar%2522%252C%2522activeTabId%2522%253A%2522clozyzjqd07y53b6hcozn80kn%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clozyzjqd07y53b6hcozn80kn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clozs6ekd00073b6h2nn3j5ox%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00073b6h2nn3j5ox%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clozs6ekd00043b6h64x52y47%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clozs6ekd00053b6hdmg6nz9l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clozs6ekd00063b6h864igdkv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522preview%2522%257D%252C%257B%2522id%2522%253A%2522clozz4vnj09y23b6hktyyi1fs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clozz4vt300abedipfaajccxk%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clozz4vnj09y23b6hktyyi1fs%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A20.266203703703695%257D) with a minimal reproduction.
The repro is showing the app with a teal box that changes it's opacity on hover.
The application seems to be working well, but the test is applying the pseudo states (like ":hover"/"_hover") without them needing to be applied.

The funny thing is, I encountered this when removing `react-scripts@4.0.3` from my project.
When I try adding this dep to the repro sandbox, it still doesn't work 😕 😳 

### System Info

```shell
System:
    OS: Linux 6.1 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 AMD EPYC
    Memory: 355.80 MB / 2.01 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 20.9.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.1.0 - /usr/local/bin/npm
  npmPackages:
    @vitejs/plugin-react: 4.1.1 => 4.1.1 
    @vitejs/plugin-react-swc: ^3.4.1 => 3.4.1 
    vite: ^4.5.0 => 4.5.0 
    vitest: 0.34.6 => 0.34.6
```

### Used Package Manager

yarn

### Link to Reproduction

[Codesandbox](https://codesandbox.io/p/sandbox/confident-fast-67vlvw?file=%2Ftsconfig.json%3A9%2C37&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clozs6ekd000a3b6huws40ft8%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clozs6ekd00033b6hzhkhigth%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clozs6ekd00073b6h2nn3j5ox%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clozs6ekd00093b6h042la2ar%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B57.833957888181416%252C42.166042111818584%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clozs6ekd00033b6hzhkhigth%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00033b6hzhkhigth%2522%252C%2522activeTabId%2522%253A%2522clozzhs1c0c2u3b6hredvibof%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clozs6ekd00023b6hl95p6b1y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clozu1jqi00e73b6h288ih2nl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clozu5xvm02233b6henjbx4ur%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FCell.tsx%2522%252C%2522id%2522%253A%2522clozu86fm038f3b6h92zw1jo7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.test.tsx%2522%252C%2522id%2522%253A%2522clozz4max09tt3b6hwb9ytrvw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fvite.config.ts%2522%252C%2522id%2522%253A%2522clozzhs1c0c2u3b6hredvibof%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FProvider.tsx%2522%252C%2522id%2522%253A%2522clp14u11g0iz63b6h999g5a8n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clp15336000033b6i7vnhz7oj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clozs6ekd00093b6h042la2ar%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00093b6h042la2ar%2522%252C%2522activeTabId%2522%253A%2522clozyzjqd07y53b6hcozn80kn%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clozyzjqd07y53b6hcozn80kn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clozs6ekd00073b6h2nn3j5ox%2522%253A%257B%2522id%2522%253A%2522clozs6ekd00073b6h2nn3j5ox%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clozs6ekd00043b6h64x52y47%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clozs6ekd00053b6hdmg6nz9l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clozs6ekd00063b6h864igdkv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522preview%2522%257D%252C%257B%2522id%2522%253A%2522clozz4vnj09y23b6hktyyi1fs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clozz4vt300abedipfaajccxk%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clozz4vnj09y23b6hktyyi1fs%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A20.266203703703695%257D)

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

1.8.9

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by KoninMikhail** [comment]
+1

**Comment by AmirTugi** [comment]
Funny thing is that this setup worked fined when using `jest` and `react-scripts`, and also when I replaced `jest` with `vitest` but still kept `react-scripts`.
I truly wonder what is it that makes it work when combining this with `react-scripts`.

**Comment by segunadebayo** [comment]
This is largely beyond the scope of Chakra. I'm not sure how we can help resolve this given it's a Vitest specific issue.
